packages/widgets/src/health-monitoring/component.tsx

```diff
@@ -0,0 +1,345 @@
+"use client";
+
+import {
+  Avatar,
+  Box,
+  Card,
+  Center,
+  Flex,
+  Group,
+  HoverCard,
+  Indicator,
+  List,
+  Progress,
+  RingProgress,
+  Text,
+  Tooltip,
+} from "@mantine/core";
+import { useElementSize, useListState } from "@mantine/hooks";
+import {
+  IconBrain,
+  IconClock,
+  IconCpu,
+  IconCpu2,
+  IconFileReport,
+  IconInfoSquare,
+  IconServer,
+  IconTemperature,
+  IconVersions,
+} from "@tabler/icons-react";
+
+import type { HealthMonitoring } from "@homarr/integrations";
+import type { TranslationFunction } from "@homarr/translation";
+import { useI18n } from "@homarr/translation/client";
+
+import type { WidgetComponentProps } from "../definition";
+import { NoIntegrationSelectedError } from "../errors";
+
+export default function HealthMonitoringWidget({
+  options,
+  integrationIds,
+  serverData,
+}: WidgetComponentProps<"healthMonitoring">) {
+  if (integrationIds.length === 0) {
+    throw new NoIntegrationSelectedError();
+  }
```

Comment:
This should be put underneath all hooks (So underneath the `useListState`. Otherwise when count is updated from 0 to 1 it will show a react error `More hooks rendered then in previous render`